src/theory/strings/theory_strings.cpp

```diff
@@ -2077,6 +2077,20 @@ void TheoryStrings::checkExtfInference( Node n, Node nr, ExtfInfoTmp& in, int ef
             do_infer = pol ? !areEqual(lit[0], lit[1])
                            : !areDisequal(lit[0], lit[1]);
           }
+          else if (lit.getKind() == OR)
```

Comment:
There are some rewrites of `str.contains` that result in an conjunction, so we should probably handle it here as well?